plugin/path_static_account.go

```diff
@@ -0,0 +1,346 @@
+package gcpsecrets
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const (
+	staticAccountStoragePrefix       = "static"
+	staticAccountPathPrefix          = "static"
+	gcpServiceAccountInferredProject = "-"
+)
+
+func pathStaticAccount(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: fmt.Sprintf("%s/%s", staticAccountPathPrefix, framework.GenericNameRegex("name")),
+		Fields: map[string]*framework.FieldSchema{
+			"name": {
+				Type:        framework.TypeString,
+				Description: "Required. Name to refer to this static account in Vault.",
+			},
+			"secret_type": {
+				Type:        framework.TypeString,
+				Description: fmt.Sprintf("Type of secret generated for this account. Defaults to %q", SecretTypeAccessToken),
+				Default:     SecretTypeAccessToken,
+			},
+			"service_account_email": {
+				Type:        framework.TypeString,
+				Description: "Email of the GCP service account to manage.",
+			},
+			"bindings": {
+				Type:        framework.TypeString,
+				Description: "Bindings configuration string.",
+			},
+			"token_scopes": {
+				Type:        framework.TypeCommaStringSlice,
+				Description: fmt.Sprintf(`List of OAuth scopes to assign to access tokens generated under this account. Ignored if "secret_type" is not "%q"`, SecretTypeAccessToken),
+			},
+		},
+		ExistenceCheck: b.pathStaticAccountExistenceCheck,
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.DeleteOperation: &framework.PathOperation{
+				Callback: b.pathStaticAccountDelete,
+			},
+			logical.ReadOperation: &framework.PathOperation{
+				Callback: b.pathStaticAccountRead,
+			},
+			logical.CreateOperation: &framework.PathOperation{
+				Callback: b.pathStaticAccountCreate,
+			},
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathStaticAccountUpdate,
+			},
+		},
+		HelpSynopsis:    pathStaticAccountHelpSyn,
+		HelpDescription: pathStaticAccountHelpDesc,
+	}
+}
+
+func pathStaticAccountList(b *backend) *framework.Path {
+	// Paths for listing role sets
+	return &framework.Path{
+		Pattern: fmt.Sprintf("%ss?/?", staticAccountPathPrefix),
```

Comment:
Is this supposed to be `%ss?/?`?

Reply:
Can we add a test case to cover this endpoint?

Reply:
I think the original intention was for this path to be `statics` in a similar vein to listing `rolesets`. I've removed the plural "s". I can add it in if you think that's more consistent.

Added test step to list static accounts in the existing tests in 1bdd252470b3b65796ced60b10d530430d44a64d

Reply:
The database engine has this under `static-roles`, would something similar or `static-rolesets` work here?

Reply:
I see that the new regex is `%s?/?`, but should it be `%s/?`

Reply:
Since this is for managing IAM service account, I'm leaning towards prefixing this with `static-account`. I see what you mean on adding the `s` to match LIST on `rolesets` though... If we have it as `static-account` I'd be okay with having `%ss/?` as the path for LIST to keep that pattern consistent within the engine. We should enforce the plural naming if we're sticking with that though instead of making that optional.

Reply:
I've changed the prefix `static-account`. I left the regex as `%ss?/?` because the path to list rolesets also has the `s` as optional (`rolesets?/?`). What do you think about that?

Reply:
I'm good with `%ss?/?`, since that's consistent with listing rolesets. Also good with `static-account` as the prefix 👍